### DISCUSSION TITLE
Up-to-date OpenThread spinel protocol specifications

### DISCUSSION THREAD

**Discussion by silabs-sebastien** [openthread/openthread#8334] [Q&A] [answered]
Hi,

I was searching online for the spinel protocol specs. My results included:

- A [GitHub repo](https://github.com/openthread/spinel-spec) that was archived in 2019 and I'm pretty sure there would've been (at least minor) updates to it in the last 3 years.
- The [openthread.io co-processor](https://openthread.io/platforms/co-processor) page that links to an [expired draft document (IETF) from 2017](https://datatracker.ietf.org/doc/html/draft-rquattle-spinel-unified).
- Various sources redirected me to the [NCP source of the main openthread repo](https://github.com/openthread/openthread/tree/main/src/ncp).

Obviously, none of these seem like quite enough to get an idea of the modern iteration of the spinel protocol currently used in OpenThread without digging into its actual implementation.

Does anyone know where I could find such a document detailing the up-to-date specifications for the spinel protocol as used by OpenThread?

Sincerely,

**Comment by jwhui** [comment, marked as answer]
The latest Spinel protocol specification is in [src/lib/spinel/spinel.h](https://github.com/openthread/openthread/blob/main/src/lib/spinel/spinel.h).

**Reply by silabs-sebastien** [reply]
Thanks @jwhui, much appreciated. Somehow I didn't think to look at this obvious location for details on the protocol! 😅
